src/commands/math/basicSymbols.ts

```diff
@@ -357,30 +423,50 @@ var Letter = P(Variable, function(_, super_) {
 
     // removeClass and delete flags from all letters before figuring out
     // which, if any, are part of an operator name
-    Fragment(l[R] || this.parent.ends[L], r[L] || this.parent.ends[R]).each(function(el) {
-      el.italicize(true).jQ.removeClass('mq-first mq-last mq-followed-by-supsub');
-      el.ctrlSeq = el.letter;
+    var lR = l && l[R];
+    var rL = r && r[L];
+
+    new Fragment(lR || this.parent.ends[L] as MQNode, rL || this.parent.ends[R] as MQNode).each(function(el) {
+      if (el instanceof Letter) {
+        el.italicize(true).jQ.removeClass('mq-first mq-last mq-followed-by-supsub');
+        el.ctrlSeq = el.letter;  
+      }
+      return undefined;
     });
 
+    let autoOpsLength = autoOps._maxLength || 0;
+    
     // check for operator names: at each position from left to right, check
     // substrings from longest to shortest
-    outer: for (var i = 0, first = l[R] || this.parent.ends[L]; i < str.length; i += 1, first = first[R]) {
-      for (var len = min(autoOps._maxLength, str.length - i); len > 0; len -= 1) {
+    outer: for (var i = 0, first = (l as MQNode)[R] || this.parent.ends[L]; first && i < str.length; i += 1, first = (first as MQNode)[R]) {
+      for (var len = min(autoOpsLength, str.length - i); len > 0; len -= 1) {
         var word = str.slice(i, i + len);
+        var last:MQNode = undefined!; // TODO - TS complaining that we use last before assigning to it
```

Comment:
I'm guessing this got resolved? I didn't see any typescript lint errors on the top of this branch.

Reply:
Nevermind, I just saw the trailing bang and looked it up, you can ignore my other question about this syntax as well.

https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci

Reply:
Telling TypeScript that your `undefined` value is, in fact, defined seems like bad practice.  Why not fix this by changing the type of `last` to `MQNode | undefined`?  If you're putting an `undefined` in there, you better be checking for it.

Reply:
I think basically all of the `!` and `as MQNode` casts are because the code was already assuming these casts implicitly. The scope of this typescript conversion was to not change behavior or structure. It was to make any implicit assumptions more explicit. We can now find those assumptions easily by grepping.

We had a second project after this that built on this work. A lot of code was restructured to remove these casts. I imagine a few casts still remain. I don't think it now makes sense to try to do much work on top of this PR. We'd already merged it locally and have heavily modified it.

Reply:
It's also possible that I didn't get the exact typings 100% correct on the first pass. It was hard to tell in many cases. I think it would make sense to revisit the casts that remain on top of the work that removes jquery. A lot more time went into honing the types.

Reply:
Then, that's not the TODO comment, I'd expect for this.  The following seems more on point:
```
// FIXME: asserting `undefined` exists to maintain legacy JS behavior; should be retype as `MQNode | undefined`
```